krs/groups.py

```diff
@@ -321,12 +321,29 @@ def main():
     parser_remove_user_group.add_argument('username', help='username of user')
     parser_remove_user_group.add_argument('group_path', help='group path (/parentA/parentB/name)')
     parser_remove_user_group.set_defaults(func=remove_user_group)
+    parser_modify = subparsers.add_parser('modify', help='modify an existing group')
+    parser_modify.add_argument('group_path', help='group path (/parentA/parentB/name)')
+    parser_modify.add_argument('--new-group-path', metavar='PATH', help='change group path (/parentA/parentB/name)')
+    parser_modify.add_argument('attrs', nargs=argparse.REMAINDER, help='space-separated name=value attributes. To assign lists, use name multiple times: foo=1 foo=2')
+    parser_modify.set_defaults(func=modify_group)
     args = vars(parser.parse_args())
 
     logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
 
     rest_client = get_rest_client()
     func = args.pop('func')
+    if 'attrs' in args:
+        attrs = {}
+        for item in args['attrs']:
+            k,v = item.split('=', 1)
+            if k in attrs:
+                if isinstance(attrs[k], list):
+                    attrs[k].append(v)
+                else:
+                    attrs[k] = [attrs[k], v]
+            else:
+                attrs[k] = v
```

Comment:
I'd suggest this (with appropriate import `from collections import defaultdict`):

```python
        attrs = defaultdict(list)
        for item in args['attrs']:
            k,v = item.split('=', 1)
            if (attrs[k] and not v) or (attrs[k] is None and v):
                raise Exception('cannot assign to attr and leave it empty')
            elif v:
                attrs[k].append(v)
            else:
                attrs[k] = None
```

Note the `None` in case you want to delete an attr.